k0776243/Midterm.py

```diff
@@ -0,0 +1,80 @@
+# Skeletons invade pirate ship.  Hero is stuck in bottom of ship and has to escape, get off the boat and to the island.  
+from sys import exit
+from random import randint
+
+class Ship(object):
+    
+    def enter(self)
+        print "You're stuck in the bottom of the ship and it's filling rapidly with water."
+        print "Choose between: a crowbar, the trapdoor key, or broccoli."
+    
+    action = raw_input("> ")
+    
+    if action == "crowbar":
+        print "You bash the crowbar into the overhead trapdoor."
+        print "You pry it open and a skeleton sees you and stabs you."
+        print "You die.  You have two lives left."
+        
+    elif action == "trapdoor key":
+        print "You find the key and unlock the trapdoor."
+        print "Once you get out in the open, you run for it, promptly fall in a puddle and off the ship."
+        print "There's a shark in the water.  You die.  You have two lives left."
+        
+    elif action == "broccoli":
+        print "You find stale broccoli."
+        print "You make friends with the shark through a window and he bashes it open.  You're free."
+        
+
+
+class Water(object):
+    
+    def __init__(self, scene_map):
+        print "You're swimming in the water."
+        print "You're going to run out of air."
+        print "Choose between swimming to the island, riding the shark to the island, or following ADHD and seeing...wait was that a blue dot?"
+        
+        action = raw_input("> ")
+        
+        if action == "swim to island":
+            print "You start breast crawling to the faraway island."
+            print "Your lungs start dragging at the cold air."
+            print "You start to sink and drown dead."
+            print "You have one life left."
+        
+        elif action == "ride shark to island":
+            print "You straddle the shark and he starts to swim."
+            print "But sharks have no sense of direction."
+            print "You go in circled til you get dizzy, fall into the water and hit your head on the boulder."
+            print "You have one life left."
+        
+        elif action == "follow ADHD":
+            print "You see something blue and swim after it."
+            print "Turns out it's a mermaid tail and so you ride the mermaid."
+            print "You have arrived at the island."
+
+class Island(Scene):
```

Comment:
Make sure that you have defined `Scene` before you do this...the way he does it in ex43 will work for you I think. It was something close to this

```python
class Scene(object):
    def enter():
        print "You need to provide your own implementation of enter"
        exit(1)
```